### PR TITLE
Point2DEmbedderIn3D adds

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
New constructor in Point2DEmbedderIn3D to explicitly orient the image plane and new shift method to avoid recomputing orientation plane. It could be useful to improve for instance mesh2heightfield (see DGtalTools PR [#429](https://github.com/DGtal-team/DGtalTools/pull/429) in DGtalTools).

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug mode.
- [x] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

**Comment by kerautret** [comment]
thanks @dcoeurjo your PR looks to have fix the GitHub actions!

File: src/DGtal/kernel/BasicPointFunctors.h [hunk 1]
```diff
@@ -464,6 +464,38 @@ namespace functors
...
+     * @param orientXaxisVector the vector determining  the 3D orientation of the image plane (from the x axis).
+     * @param anWidth the width to determine the 2d domain bounds (the resulting 2d domain will be a square of length anWidth).
+     * @param aDefautPoint the point given when the resulting point is outside the domain (default Point(0,0,0)).
+     *
+     */
+
+    Point2DEmbedderIn3D( const TDomain3D &aDomain3DImg,
+                         const Point &anOriginPoint, const typename Space::RealPoint & anNormalVector,
+                         const typename Space::RealPoint & orientXaxisVector,
+                         const typename Point::Component  &anWidth,
```
Comment (dcoeurjo) [inline src/DGtal/kernel/BasicPointFunctors.h:483]:
```suggestion
                         const typename Point::Component  &aWidth,
```
and in the code

**Review by dcoeurjo — CHANGES_REQUESTED** [review]
Thanks, just a typo

File: src/DGtal/kernel/BasicPointFunctors.h [hunk 2]
```diff
@@ -464,6 +464,38 @@ namespace functors
...
+     * @param anOriginPoint the center point given in the 3D domain.
+     * @param anNormalVector the normal vector to the 2d domain embedded in 3D.
+     * @param orientXaxisVector the vector determining  the 3D orientation of the image plane (from the x axis).
+     * @param anWidth the width to determine the 2d domain bounds (the resulting 2d domain will be a square of length anWidth).
+     * @param aDefautPoint the point given when the resulting point is outside the domain (default Point(0,0,0)).
+     *
+     */
+
+    Point2DEmbedderIn3D( const TDomain3D &aDomain3DImg,
+                         const Point &anOriginPoint, const typename Space::RealPoint & anNormalVector,
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply src/DGtal/kernel/BasicPointFunctors.h:481]: and in the code/doc

**Comment by kerautret** [comment]
> Thanks, just a typo

Thanks @dcoeurjo done  ;)

File: src/DGtal/kernel/BasicPointFunctors.h [hunk 3]
```diff
@@ -464,6 +464,38 @@ namespace functors
...
+                         const Point &aDefautPoint = Point(0,0,0)): myDomain(aDomain3DImg),
+                                                                    myDefaultPoint (aDefautPoint)
+    {
+     
+      typename Space::RealPoint uDir1;
+      uDir1 = orientXaxisVector/orientXaxisVector.norm();
+      typename Space::RealPoint uDir2;
+      uDir2[0] = uDir1[1]*anNormalVector[2]-uDir1[2]*anNormalVector[1];
+      uDir2[1] = uDir1[2]*anNormalVector[0]-uDir1[0]*anNormalVector[2];
+      uDir2[2] = uDir1[0]*anNormalVector[1]-uDir1[1]*anNormalVector[0];
```
Comment (dcoeurjo) [inline src/DGtal/kernel/BasicPointFunctors.h:493]:
```suggestion
      uDir2[0] = uDir1[1]*anNormalVector[2]-uDir1[2]*aNormalVector[1];
      uDir2[1] = uDir1[2]*anNormalVector[0]-uDir1[0]*aNormalVector[2];
      uDir2[2] = uDir1[0]*anNormalVector[1]-uDir1[1]*aNormalVector[0];
```

Comment (dcoeurjo) [reply src/DGtal/kernel/BasicPointFunctors.h:493]: +chekc the doc and the  `aWidth`

Comment (kerautret) [reply src/DGtal/kernel/BasicPointFunctors.h:493]: Thanks ;)

**Review by dcoeurjo — APPROVED** [review]

**Comment by dcoeurjo** [comment]
Thx, merged

**Pull request merged**